### PR TITLE
EA-131| EmrConceptSearchController should allow search by concept reference term code

### PR DESCRIPTION
**Pull request by ShruthiPitta**
https://issues.openmrs.org/browse/EA-131

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java [hunk 1]
```diff
@@ -43,28 +45,42 @@
...
+        List<ConceptSource> conceptSources = emrApiProperties.getConceptSourcesForDiagnosisSearch();
         Locale locale = Locale.ENGLISH;
-        List<ConceptSearchResult> conceptSearchResults = emrService.conceptSearch(query, locale, null, diagnosisSets, null, limit);
-        List<ConceptName> matchingConceptNames = new ArrayList<ConceptName>();
-        for (ConceptSearchResult searchResult : conceptSearchResults) {
-            matchingConceptNames.add(searchResult.getConceptName());
-        }
-        return createListResponse(matchingConceptNames);
+        List<ConceptSearchResult> conceptSearchResults = emrService.conceptSearch(query, locale, null, diagnosisSets, conceptSources, limit);
+        return createListResponse(conceptSearchResults, conceptSources.get(0));
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: Is see that you are passing conceptSources.get(0)
What happens when conceptSources has more than one source?

Comment (ShruthiPitta) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: @dkayiwa It is returning the ICD10 concept source converting it to list https://github.com/openmrs/openmrs-module-emrapi/blob/5296fb6c3ea2e382a14195cb00b0212f42f4762c/api/src/main/java/org/openmrs/module/emrapi/EmrApiProperties.java#L189. However we may have to refactor it to return a concept source which is configured as a global property.

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: Do you know why it is returning it to a list when it is actually one?

Comment (ShruthiPitta) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: No. I do not know. However there is another issue: https://issues.openmrs.org/browse/EA-130. As part of this we could refactor it?

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: Can we make this code in such a way that it does not assume one item in the list? That way, when it gets fixed to return more than one item, this code does not need to be changed.

Comment (ShruthiPitta) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: I think the fix will still return only one concept source. If you want me to change the method signature to return one concept source I can do that.

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: What puzzles me is why return a list when expecting only one item? Could there be something we are missing that the original author had in mind?

Comment (ShruthiPitta) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: ok. May be @djazayeri can help us here. @djazayeri  Do you know why EmrApiProperties.getConceptSourcesForDiagnosisSearch returns a list though we have are looking only for ICD10 source? Or may be you just moved the code for some refactoring?

Comment (mks-d) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: @ShruthiPitta and @dkayiwa, I asked this [here](https://talk.openmrs.org/t/ui-for-diagnosis-search/14834/9?u=mksd) on Talk.

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/EmrConceptSearchController.java]: Thanks @mks-d 😊

**Pull request merged**